feat: Output build_id for Android to proper locations

fuzzybinary opened the pull request:
### What and why?

`datadog-ci` will expect a `build_id` file at `unityLibrary/symbols` alongside the LineNumberMappings.json (which is copied there) and the `.so` files that Unity generates from its build.

During build, we output the build id that Unity gives us to both `unityLibrary/symbols/build_id' (for use with `datadog-ci`) and 'unityLibrary/src/main/assets/datadog.buildId` (where it can be picked up by the Android build process and put in the application for the Android SDK to read).

This isn't 100% equivalent to the Android Gradle Plugin build_id generation, as it's not generated on each gradle build, but I believe it should work for our purposes.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests

fuzzybinary committed:
feat: Output build_id for Android to proper locations

This isn't 100% equivalent to the Android Gradle Plugin build_id generation, as it's not generated on each gradle build, but should work for our purposes.

0xnm commented on samples/Datadog Sample/ProjectSettings/GvhProjectSettings.xml:
```diff
@@ -8,7 +8,7 @@
   <projectSetting name="Google.IOSResolver.PodfileAlwaysAddMainTarget" value="True" />
   <projectSetting name="Google.IOSResolver.PodfileEnabled" value="True" />
-  <projectSetting name="Google.IOSResolver.PodfileStaticLinkFrameworks" value="False" />
+  <projectSetting name="Google.IOSResolver.PodfileStaticLinkFrameworks" value="True" />
```
Is this change related to the PR?

fuzzybinary replied:
It's not, but a generalized build change so it got looped in here.

0xnm commented on packages/Datadog.Unity/Editor/SymbolAssemblyBuildProcess.cs:
```diff
@@ -95,25 +98,55 @@ internal void AndroidCopyMappingFile(DatadogConfigurationOptions options, string
+                {
+                    // Write the build id to the Android assets directory
+                    var androidAssetsDir = Path.Join(outputPath, "unityLibrary/src/main/assets");
+                    var androidBuildIdPath = Path.Join(androidAssetsDir, "datadog.buildId");
```
unity builds don't have a concept of variants?

fuzzybinary replied:
If it does, I'm not sure how to access them or add things to them. When you build the project from Unity, it does perform some incremental building, but then overwrites the Android project file with the new stuff.

The one place we *might* get in trouble is a new build from already exported project, but from what I can tell the major symbol files are built by that point and wouldn't change for the new android build.

fuzzybinary committed:
Fix build guid generation for Android

We don't get a build GUID until too late in the build process to use it for Android. Use our own instead.
